docs/commands-details/workload_create_update_apply.md

```diff
@@ -798,12 +802,79 @@ tanzu apps workload apply tanzu-java-web-app --git-repo https://github.com/vmwar
    12 + |      ref:
    13 + |        branch: main
    14 + |      url: https://github.com/vmware-tanzu/application-accelerator-samples
-  15 + |    subPath: tanzu-java-web-app
+   15 + |    subPath: tanzu-java-web-app
 ❓ Do you want to create this workload? [yN]:
 ```
 
 </details>
 
+### <a id="apply-output"></a> `--output`, `-o`
+
+This flag can be used to retrieve a workload right after it's applied in the specified format (`yaml`, `yml`, `json`).
+If used with `--yes` flag, all prompts are skipped and it only returns the workload definition.
+It can also be used with `--wait` or `--tail` flags in order to return the workload with its status.
+
+<details><summary>Example</summary>
+
+```bash
+tanzu apps workload apply rmq-sample-app --git-repo https://github.com/jhvhs/rabbitmq-sample --git-branch main --service-ref "rmq=rabbitmq.com/v1beta1:RabbitmqCluster:example-rabbitmq-cluster-1" --type web --output yaml
+🔎 Create workload:
+      1 + |---
+      2 + |apiVersion: carto.run/v1alpha1
+      3 + |kind: Workload
+      4 + |metadata:
+      5 + |  labels:
+      6 + |    apps.tanzu.vmware.com/workload-type: web
+      7 + |  name: rmq-sample-app
+      8 + |  namespace: default
+      9 + |spec:
+     10 + |  serviceClaims:
+     11 + |  - name: rmq
+     12 + |    ref:
+     13 + |      apiVersion: rabbitmq.com/v1beta1
+     14 + |      kind: RabbitmqCluster
+     15 + |      name: example-rabbitmq-cluster-1
+     16 + |  source:
+     17 + |    git:
+     18 + |      ref:
+     19 + |        branch: main
+     20 + |      url: https://github.com/jhvhs/rabbitmq-sample
+❓ Do you want to create this workload? [yN]: y
+👍 Created workload "rmq-sample-app"
+
+To see logs:   "tanzu apps workload tail rmq-sample-app --timestamp --since 1h"
+To get status: "tanzu apps workload get rmq-sample-app"
+
+---
+apiVersion: carto.run/v1alpha1
+kind: Workload
+metadata:
+  creationTimestamp: "2023-04-04T15:18:13Z"
+  generation: 1
+  labels:
+    apps.tanzu.vmware.com/workload-type: web
+  name: rmq-sample-app
+  namespace: default
+  resourceVersion: "184169566"
+  uid: 6588d398-b803-47e3-b31a-23d9a1a633a9
+spec:
+  serviceClaims:
+  - name: rmq
+    ref:
+      apiVersion: rabbitmq.com/v1beta1
+      kind: RabbitmqCluster
+      name: example-rabbitmq-cluster-1
+  source:
+    git:
+      ref:
+        branch: main
+      url: https://github.com/jhvhs/rabbitmq-sample
+status:
```

Comment:
Do we need this status section ?

Reply:
that's an actual output, is shown like that, I just copy-pasted it.

Reply:
I reckon we want to display the exact output of the command, so marking as resolved